src/Integration.Vsix/CFamily/CMake/CompilationDatabaseRequest.cs

```diff
@@ -0,0 +1,120 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.IO;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.CFamily.CMake;
+using SonarLint.VisualStudio.Core.CFamily;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily.CMake
+{
+    /// <summary>
+    /// Encapsulates a low-level analysis request for the CFamily compilation database entry protocol
+    /// </summary>
+    internal class CompilationDatabaseRequest : IRequest
+    {
+        private readonly CompilationDatabaseEntry databaseEntry;
+
+        public CompilationDatabaseRequest(CompilationDatabaseEntry databaseEntry, RequestContext context)
+        {
+            this.databaseEntry = databaseEntry ?? throw new ArgumentNullException(nameof(databaseEntry));
+            Context = context ?? throw new ArgumentNullException(nameof(context));
+
+            // Must have a Command or Arguments but not both
+            var hasArgs = !string.IsNullOrEmpty(databaseEntry.Arguments);
```

Comment:
The spec says we should only have one or the other. I guess it depends on what the third-party tools actually produce, and whether they have any bugs. Either way, we might as well validate that we have valid inputs.